aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -249,6 +249,18 @@ Tensor full_like(const Tensor& self, Scalar fill_value, const TensorOptions& opt
   return native::full(self.sizes(), fill_value, options);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ fill diagonal ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tensor fill_diagonal(const Tensor& self, Scalar value, int64_t offset, int64_t dim1_, int64_t dim2_) {
+  int64_t nDims = self.dim();
```

Comment:
doesn't tensor.diagonal already do these checks?

Reply:
yes. I remove these checks, and change api to match Numpy api. Thank you